catalog-rest-service/src/test/java/org/openmetadata/catalog/airflow/HttpServerExtension.java

```diff
@@ -0,0 +1,60 @@
+/*
+ *  Copyright 2022 Collate
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.openmetadata.catalog.airflow;
+
+import com.sun.net.httpserver.HttpHandler;
+import com.sun.net.httpserver.HttpServer;
+import java.net.InetSocketAddress;
+import java.net.URI;
+import java.net.URISyntaxException;
+import org.apache.http.client.utils.URIBuilder;
+import org.junit.jupiter.api.extension.AfterAllCallback;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+public class HttpServerExtension implements BeforeAllCallback, AfterAllCallback {
+
+  private static final int PORT = 9999;
```

Comment:
I don't like too much a fixed port number for this test, do you think this could have side effects if for example, two CI pipelines run at the same time in the same node? If so, I could look for an alternative to find a free port.

Reply:
Ok, updated to avoid hardcoding it by adding a minor extra code to find a free port.